From 483ff765eb7d54eb30f3793a4f00e663c01aac45 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Sun, 13 Apr 2025 23:28:09 +0300
Subject: [PATCH 1000/1000] yocto: build: Update .gitignore

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index 0bbae167bf93..83ecd90bc228 100644
--- a/.gitignore
+++ b/.gitignore
@@ -10,6 +10,7 @@
 #
 # Normal rules (sorted alphabetically)
 #
+yocto.config
 .*
 *.a
 *.asn1.[ch]
-- 
2.34.1

