From bfa8afe1ae937121620cb2c8e29775e4614dab04 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Thu, 14 Feb 2019 07:44:27 +0200
Subject: [PATCH 09/10] sn65dsi83: Kconfig: Add a missing quates

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 drivers/gpu/drm/bridge/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/bridge/Kconfig b/drivers/gpu/drm/bridge/Kconfig
index a7757f7..9e4a7b5 100644
--- a/drivers/gpu/drm/bridge/Kconfig
+++ b/drivers/gpu/drm/bridge/Kconfig
@@ -126,6 +126,8 @@ source "drivers/gpu/drm/bridge/adv7511/Kconfig"
 
 source "drivers/gpu/drm/bridge/synopsys/Kconfig"
 
+source "drivers/gpu/drm/bridge/sn65dsi83/Kconfig"
+
 config DRM_ITE_IT6263
 	tristate "ITE IT6263 LVDS/HDMI bridge"
 	depends on OF
@@ -134,6 +136,4 @@ config DRM_ITE_IT6263
 	---help---
 	  ITE IT6263 bridge chip driver.
 
-source "drivers/gpu/drm/bridge/sn65dsi83/Kconfig
-
 endmenu
-- 
1.9.1

