From 77f5878a96f41f586d94515c6531eacde9998b80 Mon Sep 17 00:00:00 2001
From: Kirill Kapranov <kirill.kapranov@compulab.co.il>
Date: Sun, 21 Jun 2020 22:08:06 +0300
Subject: [PATCH 42/99] env: Add to kernel command line rw rootwait

Signed-off-by: Kirill Kapranov <kirill.kapranov@compulab.co.il>
---
 include/configs/ucm-imx8m-mini.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/configs/ucm-imx8m-mini.h b/include/configs/ucm-imx8m-mini.h
index 1d1108a6b9..c153f61ea9 100644
--- a/include/configs/ucm-imx8m-mini.h
+++ b/include/configs/ucm-imx8m-mini.h
@@ -134,11 +134,11 @@
 	"mmcroot=" CONFIG_MMCROOT " rootwait rw\0" \
 	"mmcautodetect=yes\0" \
 	"emmc_ul=setenv iface mmc; setenv dev 1; setenv part 1;" \
-	"setenv bootargs console=${console} root=/dev/mmcblk2p2;\0" \
+	"setenv bootargs console=${console} root=/dev/mmcblk2p2 rw rootwait;\0" \
 	"sd_ul=setenv iface mmc; setenv dev 0; setenv part 1;" \
-	"setenv bootargs console=${console} root=/dev/mmcblk1p2;\0" \
+	"setenv bootargs console=${console} root=/dev/mmcblk1p2 rw rootwait;\0" \
 	"usb_ul=usb start; setenv iface usb; setenv dev 0; setenv part 1;" \
-	"setenv bootargs console=${console} root=/dev/sda2;\0" \
+	"setenv bootargs console=${console} root=/dev/sda2 rw rootwait;\0" \
 	"ulbootscript=load ${iface} ${dev}:${part} ${loadaddr} ${script};\0" \
 	"ulimage=load ${iface} ${dev}:${part} ${loadaddr} ${image}\0" \
 	"ulfdt=if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
-- 
2.11.0

