From 7328ca230f8f8af35734a42a30554860e2f36240 Mon Sep 17 00:00:00 2001
From: Kirill Kapranov <kirill.kapranov@compulab.co.il>
Date: Mon, 19 Jul 2021 01:02:48 +0300
Subject: [PATCH 43/91] mcm-imx8m-mini: Set EXTRAVERSION

Signed-off-by: Kirill Kapranov <kirill.kapranov@compulab.co.il>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0451f445f6..22d2230a6f 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@
 VERSION = 2021
 PATCHLEVEL = 04
 SUBLEVEL =
-EXTRAVERSION =
+EXTRAVERSION = -mcm-imx8m-mini-1.0
 NAME =
 
 # *DOCUMENTATION*
-- 
2.17.1

