From 168cb2c5f6c8ef4bccd70243d9da2b954c132520 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Thu, 11 Jun 2020 12:04:00 +0300
Subject: [PATCH 44/46] Makefile: Clean up EXTRAVERSION

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 73aec45a9b..225fa6828c 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@
 VERSION = 2018
 PATCHLEVEL = 03
 SUBLEVEL =
-EXTRAVERSION = -mcm-imx8m-mini-1.0
+EXTRAVERSION =
 NAME =
 
 # *DOCUMENTATION*
-- 
2.11.0

