From 60e9c84cf4e51791d2880fd26bd1d15534effb85 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Sat, 4 Jul 2020 09:58:56 +0300
Subject: [PATCH 53/53] mcm-imx8m-mini: Fix basic support #3

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 include/configs/mcm-imx8m-mini.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/mcm-imx8m-mini.h b/include/configs/mcm-imx8m-mini.h
index d38ab70433..147d34ec92 100644
--- a/include/configs/mcm-imx8m-mini.h
+++ b/include/configs/mcm-imx8m-mini.h
@@ -126,7 +126,7 @@
 	"fdt_addr=0x43000000\0"			\
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fdt=yes\0" \
-	"fdt_file="CONFIG_DEFAULT_DEVICE_TREE".dtb\0" \
+	"fdt_file="CONFIG_DEFAULT_DTB"\0" \
 	"initrd_addr=0x43800000\0"		\
 	"initrd_high=0xffffffffffffffff\0" \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
-- 
2.11.0

