From 02c3d4fda7e2fb7c9806e40e34251e1c0d7c2c7c Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Wed, 5 Aug 2020 00:22:29 +0300
Subject: [PATCH 77/99] mcm-imx8m-mini: spl: Add board_fit_config_name_match()

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 board/compulab/mcm-imx8m-mini/spl.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/board/compulab/mcm-imx8m-mini/spl.c b/board/compulab/mcm-imx8m-mini/spl.c
index d6ece8ef3e..1a208e09a4 100644
--- a/board/compulab/mcm-imx8m-mini/spl.c
+++ b/board/compulab/mcm-imx8m-mini/spl.c
@@ -7,3 +7,13 @@
 int spl_board_private_init(void) {
 	return 0;
 }
+
+#ifdef CONFIG_SPL_LOAD_FIT
+int board_fit_config_name_match(const char *name)
+{
+	/* Just empty function now - can't decide what to choose */
+	debug("%s: %s\n", __func__, name);
+
+	return 0;
+}
+#endif
-- 
2.11.0

