From 020ac05923a9fbeb54ac2fc0f5a07104ec2087a5 Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Sun, 16 Aug 2020 19:03:16 +0300
Subject: [PATCH 095/111] imx8mm: enable prompt update in sub-projects

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 include/configs/cpl-imx8m-mini.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/configs/cpl-imx8m-mini.h b/include/configs/cpl-imx8m-mini.h
index dd8120f4dc..90ca1e6138 100644
--- a/include/configs/cpl-imx8m-mini.h
+++ b/include/configs/cpl-imx8m-mini.h
@@ -203,8 +203,6 @@
 #define CONFIG_MXC_UART_BASE		UART3_BASE_ADDR
 
 /* Monitor Command Prompt */
-#undef CONFIG_SYS_PROMPT
-#define CONFIG_SYS_PROMPT		"u-boot=> "
 #define CONFIG_SYS_PROMPT_HUSH_PS2     "> "
 #define CONFIG_SYS_CBSIZE              2048
 #define CONFIG_SYS_MAXARGS             64
-- 
2.11.0

