From ecebd4410e5de8fd3f438733b8923f2331b5ca28 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Wed, 9 Jun 2021 11:16:27 +0300
Subject: [PATCH] cpl-imx8m-mini: Fix CONFIG_MFG_ENV_SETTINGS

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 include/configs/cpl-imx8m-mini.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/cpl-imx8m-mini.h b/include/configs/cpl-imx8m-mini.h
index 90ca1e6138..536f9cba21 100644
--- a/include/configs/cpl-imx8m-mini.h
+++ b/include/configs/cpl-imx8m-mini.h
@@ -94,8 +94,8 @@
 		"\0" \
 	"initrd_addr=0x43800000\0" \
 	"initrd_high=0xffffffff\0" \
-	"emmc_dev=1\0" \
-	"sd_dev=0\0" \
+	"emmc_dev=2\0" \
+	"sd_dev=1\0" \
 	"bootcmd_mfg=run mfgtool_args;  if iminfo ${initrd_addr}; then "\
 					   "booti ${loadaddr} ${initrd_addr} ${fdt_addr};"\
 					"else echo \"Run fastboot ...\"; fastboot 0; fi\0" \
-- 
2.11.0

